docs/BENCHMARK.md

```diff
@@ -0,0 +1,93 @@
+# Addition of `VisitedPcs` trait
+
+Current blockifier doesn't store the complete vector of visited program counters
+for each entry point call in an invoke transaction. Instead, visited program
+counters are pushed in a `HashSet`. This is a limiting factor to perform
+profiling operations which require record of the full trace returned by
+`cairo-vm`. More flexibility is added to the blockifier with the introduction of
+trait `VisitedPcs` which allows the user to process visited program counters in
+the most suitable way for the task.
+
+## Existing code
```

Comment:
"Existing Code" ceases to mean anything once this is merged. You should refer to a specific commit, and even link to the file in question (github link as you can link to a specific commit).